### DISCUSSION TITLE
Linux. I installed successfully homebrew but suddenly disapears

### DISCUSSION THREAD

**Discussion by josias-guerrero** [Homebrew/discussions#4465] [Linux] [answered]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: 763c41f006ab9b44261e313b0aaa24a047431b06
Last commit: 4 days ago
Core tap JSON: 28 Apr 17:03 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.10 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: 16-core 64-bit unknown_0x6_0xa5
Clang: N/A
Git: 2.40.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.0-71-generic x86_64 GNU/Linux
OS: Linux Mint 21.1 (vera)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A


Closing and opening another tab: 

It says that is not found
```

### Output of `brew doctor`

```shell
Your system is ready to brew.

Again, closing and opening another tab says that is not found.
```

### Description of issue

the thing is that I installed and used it perfectly, I even installed oh-my-posh using it, but when I try to close the tab and open another tab It says that is not found the brew command, so I tried to install again homebrew and it works again, but closing again the tab and opening another it says the same, that the command is not found, I did it like three times and is the same.

**Comment by ZhongRuoyu** [comment]
Did you follow the installer's instructions and add the following to your shell profile (usually `~/.bash_profile` Bash, `~/.zprofile` for Zsh)?

```
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
```

**Reply by josias-guerrero** [reply]
yup, i did it, actually I checked the profile code and it is in there 
![image](https://user-images.githubusercontent.com/119614792/235218269-b61df56c-04b6-4c88-8871-56c38c619b63.png)

**Reply by josias-guerrero** [reply]
and, executing the command that u tell me, it works again. But in the .profile it writes again the same line, but opening another time it says the same
![image](https://user-images.githubusercontent.com/119614792/235218792-faed520f-47d4-4276-b969-37d3e56ab238.png)

**Reply by ZhongRuoyu** [reply, marked as answer]
Does the following command output show you a path to `zsh`?

```
echo "$SHELL"
```

If so, you may try to log out and log back in again. That should solve the problem.

Otherwise, you may need to add the line to that shell's profile instead. Usually, that's `~/.profile`.

**Reply by josias-guerrero** [reply]
Thanks, it worked, reloading was the solution haha.
